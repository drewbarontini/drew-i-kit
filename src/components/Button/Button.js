import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { ifProp } from 'styled-tools';

import { hover } from '../../lib/hover';
import { theme, palette } from '../../lib/props';
import { responsive } from '../../lib/responsive';

const Button = styled.button`
  background-color: ${ifProp(
    'primary',
    palette('primary.base'),
    'transparent'
  )};
  border: 0;
  border: 2px solid ${ifProp('primary', 'transparent', palette('fg'))};
  border-radius: ${theme('borders.radius.base')}
  color: ${ifProp('primary', palette('white'), palette('fg'))};
  cursor: pointer;
  display: inline-block;
  font-size: ${theme('fonts.sizes.xs')};
  font-weight: ${theme('fonts.weights.bold')};
  letter-spacing: 1px;
  line-height: 3;
  padding: 0 ${theme('spacing.base')};
  text-transform: uppercase;
  transition: ${theme('transition.base')};

  ${hover(css`
    background-color: ${ifProp('primary', palette('fg'), 'transparent')};
    border-color: ${ifProp('primary', 'transparent', palette('primary.base'))};
    color: ${ifProp('primary', palette('white'), palette('primary.base'))};
  `)}

  ${responsive};
`;

Button.propTypes = {
  primary: PropTypes.bool,
};

Button.displayName = 'Button';

export default Button;
