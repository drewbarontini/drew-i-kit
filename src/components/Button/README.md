# Button

## Usage

```jsx
import { Button } from 'drew-i-kit';

<Button>My Button</Button>
```

```jsx
import { Button } from 'drew-i-kit';

<Button primary>My Button</Button>
```

## Props

- `primary` (**Boolean**, optional)