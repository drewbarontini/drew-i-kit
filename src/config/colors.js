import { paletteFor } from '../lib';

const colors = {
  ...paletteFor({
    area: 'primary',
    base: '#3576a7',
    darker: ['#152f43', '#1d415c', '#255375', '#2d648e'],
    lighter: ['#538bb4', '#729fc1', '#90B4cf', '#aeC8dc'],
  }),
  ...paletteFor({
    area: 'neutral',
    base: '#737373',
    darker: ['#0b0b0b', '#1d1d1d', '#2e2e2e', '#3f3f3f', '#515151', '#626262'],
    lighter: ['#888888', '#9d9d9d', '#b2b2b2', '#c7c7c7', '#dcdcdc', '#f1f1f1'],
  }),
  accent: {
    ...paletteFor({
      area: 'green',
      base: '#8acb88',
      darker: ['#233322', '#375136', '#4c704b', '#618e5f', '#75ad74'],
      lighter: ['#9cd39a', '#addbac', '#bfe2be', '#d0eacf', '#e2f2e1'],
    }),
    ...paletteFor({
      area: 'yellow',
      base: '#ffc527',
      darker: ['#40310a', '#664f10', '#8c6c15', '#b38a1b', '#d9a721'],
      lighter: ['#ffce47', '#ffd668', '#ffdf88', '#ffe8a9', '#fff1c9'],
    }),
    ...paletteFor({
      area: 'red',
      base: '#fe5f55',
      darker: ['#401815', '#662622', '#8c342f', '#b2433b', '#d85148'],
      lighter: ['#fe776f', '#fe8f88', '#fea7a2', '#ffbfbb', '#ffd7d5'],
    }),
  },
  bg: '#fff',
  fg: '#272727',
  white: '#fff',
  black: '#272727',
  border: '#dcdcdc',
};

export default colors;
