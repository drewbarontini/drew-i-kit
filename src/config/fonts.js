const fonts = {
  family: {
    base:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    heading:
      "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen-Sans, Ubuntu, Cantarell, 'Helvetica Neue', sans-serif",
    monospace: "'Monaco', monospace",
  },
  lineHeight: {
    base: '1.5',
    heading: '1.2',
  },
  sizes: {
    base: '14px',
    xxs: '8px',
    xs: '10px',
    s: '12px',
    l: '16px',
    xl: '24px',
    h1: '30px',
    h2: '26px',
    h3: '22px',
    h4: '18px',
  },
  weights: {
    light: 200,
    normal: 400,
    semiBold: 600,
    bold: 800,
  },
};

export default fonts;
