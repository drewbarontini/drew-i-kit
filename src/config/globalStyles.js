import { css } from 'styled-components';

import { palette, theme } from '../lib/props';
import { responsive } from '../lib/responsive';

const globalStyles = css`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  body {
    background-color: ${palette('bg')};
    color: ${palette('fg')};
    font-family: ${theme('fonts.family.base')};
    font-size: ${theme('fonts.sizes.base')};
    line-height: ${theme('fonts.lineHeight.base')};
    margin: ${theme('spacing.l')};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${theme('fonts.family.heading')};
    line-height: ${theme('fonts.lineHeight.heading')};
    margin: 0;
    margin-bottom: ${theme('spacing.s')};
  }

  img {
    display: block;
    max-width: 100%;
  }

  ${responsive};
`;

export default globalStyles;
