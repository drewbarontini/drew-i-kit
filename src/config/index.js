export { default as cssProps } from './cssProps';
export { default as icons } from './icons';
export { default as theme } from './theme';
