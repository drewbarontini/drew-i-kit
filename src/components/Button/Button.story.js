import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from '.';

storiesOf('Button', module)
  .add('default', () => <Button>Default Button</Button>)
  .add('primary', () => <Button primary>Primary Button</Button>);
