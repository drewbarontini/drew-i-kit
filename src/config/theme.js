import colors from './colors';
import fonts from './fonts';
import icons from './icons';

const theme = {
  borders: {
    sizes: {
      base: '1px',
      l: '2px',
    },
    radius: {
      base: '3px',
    },
  },
  breakpoints: {
    sizes: {
      s: 380,
      m: 768,
      l: 900,
      xl: 1100,
    },
    query: 'max-width',
    unit: 'px',
  },
  colors: {
    ...colors,
  },
  fonts: {
    ...fonts,
  },
  icons: {
    ...icons,
  },
  shadows: {
    base: '0 3px 3px rgba(0, 0, 0, 0.25);',
  },
  sizing: {
    base: '900px',
    s: '400px',
    l: '1200px',
  },
  spacing: {
    base: '20px',
    xs: '5px',
    s: '10px',
    l: '40px',
    xl: '60px',
  },
  transition: {
    base: '0.3s ease-in-out',
  },
};

export default theme;
