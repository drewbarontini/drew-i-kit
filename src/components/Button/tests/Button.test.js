import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import { theme } from '../../../config';

import Button from '../';

describe('<Button />', () => {
  it('should render properly', () => {
    const { container } = render(<Button theme={theme}>My Button</Button>);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild.tagName.toLowerCase()).toEqual('button');
    expect(container.firstChild).toHaveTextContent('My Button');
    expect(container.firstChild).toHaveStyleRule(
      'background-color',
      'transparent'
    );
    expect(container.firstChild).toHaveStyleRule(
      'border',
      `2px solid ${theme.colors.fg}`
    );
    expect(container.firstChild).toHaveStyleRule('color', theme.colors.fg);
  });
});
