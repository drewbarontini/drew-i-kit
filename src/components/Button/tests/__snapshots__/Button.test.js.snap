// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
.c0 {
  border: 0;
  border-style: solid;
  cursor: pointer;
  padding-bottom: 0;
  padding-top: 0;
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
  background-color: transparent;
  border-color: #272727;
  border-radius: 3px;
  border-width: 2px;
  color: #272727;
  display: inline-block;
  font-size: 10px;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  line-height: 3;
  padding-left: 20px;
  padding-right: 20px;
  text-transform: uppercase;
}

.c0:active,
.c0:focus,
.c0:hover {
  background-color: transparent;
  border-color: #3576a7;
  color: #3576a7;
}

<button
  class="c0"
>
  My Button
</button>
`;
